single_kernel_mongo/managers/ldap.py

```diff
@@ -0,0 +1,175 @@
+        """Returns True if the integration to ldap is valid."""
+        return (self.state.ldap_relation is not None) and not self.state.is_role(MongoDBRoles.SHARD)
+
+    def on_ldap_ready(self, relation: Relation) -> None:
+        """Runs when LDAP is ready."""
+        self.assert_pass_hook_checks()
+
+        self.charm.status_manager.set_and_share_status(MaintenanceStatus("Configuring LDAP"))
+
+        ldap_data = self.ldap_requirer.consume_ldap_relation_data(relation=relation)
+        if not ldap_data:
+            logger.info("Waiting for LDAP data.")
+            raise WaitingForLdapDataError("Waiting for LDAP data.")
+
+        if not ldap_data.ldaps_urls:
+            logger.error("You must configure Glauth with LDAPS support")
+            raise LDAPSNotEnabledError("LDAPS not enabled.")
+
+        if self.charm.unit.is_leader():
+            self.state.ldap.set_from(ldap_data)
+
+        self.restart_when_ready()
+
+    def restart_when_ready(self):
+        """Restarts when we are ready."""
+        if not self.state.ldap_relation:
+            logger.info(f"Not restarting, missing the `{ExternalRequirerRelations.LDAP}` relation")
+            return
+        if not self.state.ldap_cert_relation:
+            logger.info(
+                f"Not restarting, missing the `{ExternalRequirerRelations.LDAP_CERT}` relation"
+            )
+            return
+
+        if self.state.ldap.is_ready():
+            logger.info("Restarting mongodb server for LDAP integration")
+            self.dependent.restart_charm_services()
+            self.charm.status_manager.set_and_share_status(ActiveStatus())
+
+    def on_ldap_unavailable(self) -> None:
+        """Runs when the LDAP integration is broken."""
+        self.state.ldap.clean_databag()
+
+        self.dependent.restart_charm_services()
+
+    def on_certificate_available(self, certificate: str, ca: str, chain: list[str]):
+        """Runs when we receive the LDAP certificates."""
+        self.assert_pass_hook_checks()
+        self.state.ldap.set_certificates(certificate, ca, chain)
+
+        full_chain = "\n".join(chain)
+        self.dependent.save_ca_cert_to_trust_store(TrustStoreFiles.LDAP, full_chain)
+
+        self.restart_when_ready()
+
+    def on_certificate_removed(self):
+        """Runs when the certificate is removed."""
+        self.state.ldap.set_certificates(None, None, None)
+        self.dependent.remove_ca_cert_from_trust_store(TrustStoreFiles.LDAP)
+
+        self.dependent.restart_charm_services()
+
+    def get_status(self) -> StatusBase | None:
+        """Generates the status of a unit based on its status reported by mongod."""
+        if self.state.ldap_relation is None and self.state.ldap_cert_relation is None:
+            return None
+
+        if self.state.is_role(MongoDBRoles.SHARD):
+            return BlockedStatus("Cannot integrate LDAP with shard.")
+
+        if self.state.ldap_relation is not None and self.state.ldap_cert_relation is None:
```

Comment:
The first part of the condition is implied
```suggestion
        if self.state.ldap_cert_relation is None:
```